spec.md

```diff
@@ -1495,6 +1613,106 @@ For success responses, the following fields are defined:
 }
 ```
 
+## Adapting Service Binding Credentials
+
+If the endpoints of a Service Instance are from an Application point of view
+are different to the endpoints provided in the binding credentials, then the
+binding credentials MUST be adapted before the Platform provides them to
+the Application.
+
+The Platform requests the corrected credentials from the Service Broker by
+providing the original credentials and a table that maps the original
+endpoints to the new endpoints.
+
+### Request
+
+##### Route
+
+`PUT /v2/service_instances/:instance_id/service_bindings/:binding_id/credentials`
+
+`:instance_id` MUST be the ID of a previously provisioned Service Instance.
+
+`:binding_id` MUST be the ID of a previously provisioned Service Binding for that
+instance.
+
+#### Headers
+
+The following HTTP Headers are defined for this operation:
+
+| Header | Type | Description |
+| --- | --- | --- |
+| X-Broker-API-Version* | string | See [API Version Header](#api-version-header). |
+
+\* Headers with an asterisk are REQUIRED.
+
+#### Body
+| Request Field | Type | Description |
+| --- | --- | --- |
+| credentials* | object | The free-form hash of credentials that the Platform has received from the Service Broker. |
+| endpoint_mappings* | array of [EndpointMapping](#endpoint-mapping-object) objects | A list of endpoint mapping. |
+
+\* Headers with an asterisk are REQUIRED.
+
+##### Endpoint Mapping Object
+
+| Request Field | Type | Description |
+| --- | --- | --- |
+| source* | string | The endpoint on the service side. The format of the strings is `<hostname or IP>:<port>`. |
```

Comment:
We would propose to have a struct of host and port instead of the concatenated strings also in the endpoint mapping object.